framework/base/InvalidParamException.php

```diff
@@ -12,6 +12,7 @@
  *
  * @author Qiang Xue <qiang.xue@gmail.com>
  * @since 2.0
+ * @deprecated 2.0.14 Use [[InvalidArgumentException]]
  */
 class InvalidParamException extends \BadMethodCallException
```

Comment:
Should now extend `InvalidArgumentException` else it would break backwards compatibility.

Reply:
Exceptions are often checked with `instanceof`.

Reply:
`InvalidArgumentException` should extend `InvalidParamException` to keep BC - only then `catch (InvalidParamException $e)` will work as before.

Reply:
Doh, not sure how I forgot to do that – it was the plan.

Reply:
@samdark  is it okay that `InvalidParamException` is now deprecated and suggests to use `InvalidArgumentException` in the doc block and `InvalidArgumentException` is extending  `InvalidParamException` already deprecated exception type?

Reply:
@bologer that was discussed here: https://github.com/yiisoft/yii2/pull/15633/commits/7f4bd69dfb07214425297878e0b886a9ea364244#r167403068

Reply:
@bologer that was discussed here: https://github.com/yiisoft/yii2/pull/15633/commits/7f4bd69dfb07214425297878e0b886a9ea364244#diff-7a2fbae94c890a0d999b38f8a82fe81eR485